docs/libraries/libraries-with-cli.md

```diff
@@ -0,0 +1,572 @@
+# Writing Libraries with Cross Platform Tools
+
+By [Phillip Carter](https://github.com/cartermp)
+
+**Some details are subject to change as the toolchain evolves.**
+
+Targeting .NET Core for new libraries can be done a number of ways, depending on the platform(s) you wish to support.  If you cannot or choose not to use Visual Studio to build and test new libraries or wish to use the command line tooling on any Operating System, then this guide is for you.
+
+* [Prerequisities](#prerequisites)
+* [How to target .NET Core](#how-to-target-net-core)
+* [How to target both .NET Core and Older .NET Framework Versions](#how-to-target-both-net-core-and-older-net-framework-versions)
+* [How to target a Portable Class Library (PCL)](#how to target a portable class-library-pcl)
+* [How to cross-compile for .NET Core and .NET Framework](#how-to-cross-compile-for-net-core-and-net-framework)
+* [How to test libraries on .NET Core](#how-to-test-libraries-on-net-core)
+* [How to create a NuGet Package with your Library](#how-to-create-a-nuget-package-with-your-library)
+
+## Prerequisites
+
+You must have .NET Core installed on your machine.  This guide will focus on command line tooling because they work cross-platform and are what Visual Studio uses internally.  You have two options:
+
+* [DNX](http://aspnet.readthedocs.org/en/latest/getting-started/index.html), which has shipped with Visual Studio 2015 Update 1
+
+* [.NET CLI](http://dotnet.github.io/getting-started/), the tooling which will eventually replace DNX
+
+Either toolchain can be used to build and publish libraries today.
+
+The sections of this document dealing with older frameworks or Portable Class Libraries (PCLs) require Windows and the .NET Framework installed.  The best way to do this is to [install Visual Studio](https://www.visualstudio.com/en-us/visual-studio-homepage-vs.aspx).  Alternatively, you can [install .NET Framework 4.6](https://www.microsoft.com/en-us/download/details.aspx?id=48130) as a standalone package.  This will contain all the dependencies required for targeting older frameworks and PCLs you may wish to support.
+
+## How to target .NET Core
+
+For the purposes of building a library, targeting ".NET Core" means targeting the .NET Standard Platform.  To see what that means exactly, see [.NET Platform Standard](https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/standard-platform.md).
+
+A distilled version is this:
+
+You need to pick a version of `netstandardX.X` (where `X.X` is a version number) to add to your `project.json` file.
+
+1. If you don't care about backwards compatibility, target `netstandard1.4`.
+2. If you care about compatibility with any of the following:
```

Comment:
Compatibility is the wrong word here. We typically think of compatibility in terms of breaking changes and 'targeting' in terms of frameworks.